common/ipc-client.c

```diff
@@ -22,9 +22,13 @@ char *get_socketpath(void) {
 	size_t line_size = 0;
 	FILE *fp = popen("sway --get-socketpath 2>/dev/null", "r");
 	if (fp) {
-		getline(&line, &line_size, fp);
+		ssize_t nret = getline(&line, &line_size, fp);
 		pclose(fp);
-		if (line && *line) {
+		if (nret > 0) {
+			// remove trailing newline, if there is one
+			if (line[nret - 1] == '\n') {
```

Comment:
To quote the man page:
> On success, getline() and getdelim() return the number of characters read, including the delimiter character, but not including the terminating null byte.

So, if successful, `line[nret]` is `\0`, and if the line ended with a newline, rather than EOF, `line[nret -1]` is `\n`

Reply:
Oh, you're right, sorry. I read "including the delimiter" from the POSIX page and thought it was "including the terminating NULL character".